docs/motivation.md

```diff
@@ -0,0 +1,550 @@
+This document aims to cover the parts of the type system that are similar to (though in some places diverge in interesting ways from) other languages.
+
+Verona is an object oriented reference capability language with structural
+subtyping and bounded polymorphism. This document aims to describe the aspects
+of the type system that are independent from both the concurrency model and
+reference capabilities.
+
+## Structural type system
+
+A type system is all about managing a complexity budget. Make it too complicated
+and programmers will find it difficult to use and be less productive. Make it too
+simple and it will give you less guarantees to lean on as a programmer.
+
+We opt to use a structural type system since we beleive it simplifies some parts
+of reasoning, e.g. we don't need co/contra-variance notation on type parameters.
+This allows us to spend this complexity elsewhere, e.g. introducing reference
+capabilities to enforce isolation and region topology.
+
+We do however treat classes nominally and have opted to not have inheritance
+subtyping. This means that there is never any subtyping between two distinct
+classes.
+
+### Traits
+Verona is a structural type system based on traits, i.e. types can be described
+using combination of methods on an object. E.g. we can describe a type
+implementing a `toString` method as
+```verona
+type ToString = {
+    toString(self: Self): String
+}
+```
+A type describing numbers implementing addition and subtraction can be described with
+```verona
+type Number = {
+    add(self: Self, other: Number): Number
+    sub(self: Self, other: Number): Number
+}
+```
+Traits also support default implementation as described below in section [Trait Inheritance](#trait-inheritance).
+
+
+### Conjunction types
+We can easily combine traits using conjunctions, so if we, e.g., want to describe
+something implementing both `ToString` and `Number`, this can be written as
+```verona
+type ToStringNumber = ToString & Number
+```
+
+This gives us a flexible way to define combinations of types on the fly, with an
+intuitive syntax. For example, if we want to describe the type signature of a function
+`f` printing two numbers and then returning the result of adding them
+together, this would be
+```verona
+f(n1: Number & ToString, n2: Number & ToString) : Number
+// TODO add ToStringNumber version
+```
+
+In contrast, in a nominal system like Java, we would have to define a new
+interface `ToStringNumber` that extends both `Number` and `ToString`:
+```java
+interface ToStringNumber extends Number, ToString { ... }
+
+class C {
+    static Number f(ToStringNumber n1, ToStringNumber n2) { ... }
+}
+```
+Furthermore we would have to change the definition of every class that we want
+to implement this new trait.
+
+In Verona, a trait with multiple methods can be further simplified by breaking
+up each method into what we call an "atomic" trait and combining them with a
```

Comment:
I would not call this an "atomic" trait. Avoid using any concurrency words in text that isn't about concurrency. Also avoid adding terminology when explaining terminology.